### PR TITLE
feat: allow to use libkrun and applehv machines starting from 5.2.0

### LINKED ISSUE #8084: Allow to create libkrun machines on MacOS if Podman 5.2+ is installed
Implementation proposal:

- when listing machines, list both applehv and libkrun machines
- when creating a machine use CONTAINERS_MACHINE_PROVIDER to switch between providers
- the machine provider type select box should be displayed only on MacOS and if Podman 5.2+ is detected

### PR DESCRIPTION
**Pull request by lstocchi**
### What does this PR do?

This PR allows to manage both applehv and libkrun machines at the same time starting from 5.2.0 .
Even though a provider is defined in the `containers.conf` you should be able to work with both applehv and libkrun machines (so create, start, stop, remove, edit ... actions).

The only problem is that libkrun only supports max 8 cpus. If you create a machine with 8+ cpus, you are able to create it but it will always fail to start. So, before creating a machine, I reduce the cpu value to 8 if it's larger. It would be good to edit the maximum value at the UI level but I didn't find an easy way so far. Any idea is appreciated.

If Podman is less than 5.1.0 or you are not on a Mac it should work as before.

In draft as I still have to try it on Windows and write tests but it's ready if you want to start playing with it

### Screenshot / video of UI

https://github.com/user-attachments/assets/fd13ca3c-a075-40a2-86fc-e03b325b966f

### What issues does this PR fix or reference?

it resolves #8084 

### How to test this PR?

1. edit the value here with the Podman version you are using https://github.com/containers/podman-desktop/compare/main...lstocchi:i8084?expand=1#diff-c885162dd627b7cd812c83544b19b9b166179fec79ddbc13273fede7c2566791R1792
2. you should see and being able to use both applehv and libkrun machines. When using the cli you will only have visibility on the current provider set.

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: extensions/podman/src/extension.ts [hunk 1]
```diff
@@ -297,13 +305,16 @@ export async function updateMachines(
...
-            machineName,
-          ]);
+          const podmanMachineInfo = podmanMachinesInfo.get(machineName);
+          const { stdout: socket } = await extensionApi.process.exec(
+            getPodmanCli(),
+            ['machine', 'inspect', '--format', '{{.ConnectionInfo.PodmanSocket.Path}}', machineName],
+            {
+              env: {
+                CONTAINERS_MACHINE_PROVIDER: podmanMachineInfo?.vmType ?? '',
+              },
```
Comment (benoitf) [inline extensions/podman/src/extension.ts]: this code is repeating too many times that I think we may miss the right call.

I guess we would need a custom exec function for podman extension adding automatically that env variable

Comment (lstocchi) [reply extensions/podman/src/extension.ts]: Done

File: packages/renderer/src/lib/preferences/PreferencesResourcesRendering.svelte [hunk 2]
```diff
@@ -530,8 +531,12 @@ function hasAnyConfiguration(provider: ProviderInfo) {
...
                 updateConnectionStatus={updateContainerStatus}
                 addConnectionToRestartingQueue={addConnectionToRestartingQueue} />
-              <div class="mt-1.5 text-gray-900 text-[9px]" aria-label="Connection Version">
-                <div>{provider.name} {provider.version ? `v${provider.version}` : ''}</div>
+              <div class="mt-1.5 text-gray-900 text-[9px] flex justify-between">
+                <div aria-label="Connection Version">
+                  {provider.name}
+                  {provider.version ? `v${provider.version}` : ''}
+                </div>
+                <div aria-label="Connection Type">{container.vmType ? capitalize(container.vmType) : ''}</div>
```
Comment (jeffmaury) [inline packages/renderer/src/lib/preferences/PreferencesResourcesRendering.svelte:539]: Only capitalize the first letter that looks strange to me ?

**Comment by jeffmaury** [comment]
Can't make it to work: I have 5.2.0-rc2 installed on my Mac-M2, I can see the VMtype but can't choose the VM type while create a Podman machine

**Comment by jeffmaury** [comment]
> Can't make it to work: I have 5.2.0-rc2 installed on my Mac-M2, I can see the VMtype but can't choose the VM type while create a Podman machine

My bad, I did not follow the trick for the version check

**Review by jeffmaury — APPROVED** [review]
LGTM. Tested on Mac Pro M2.

Should'nt we have the VM type after the name of the machine ?

File: packages/renderer/src/lib/preferences/PreferencesResourcesRendering.svelte (hunk 2 above)
Comment (lstocchi) [reply packages/renderer/src/lib/preferences/PreferencesResourcesRendering.svelte:539]: Would you prefer to have the name lowercase?

File: extensions/podman/src/podman-install.ts [hunk 3]
```diff
@@ -209,9 +215,9 @@ export class PodmanInstall {
     const machinesRunning: MachineJSON[] = [];
     try {
       const machineListOutput = await getJSONMachineList();
-      const machines = JSON.parse(machineListOutput.stdout) as MachineJSON[];
+      console.log(JSON.stringify(machineListOutput));
```
Comment (benoitf) [inline extensions/podman/src/podman-install.ts]: to be removed I suppose ?

**Comment by lstocchi** [comment]
> LGTM. Tested on Mac Pro M2.
> 
> Should'nt we have the VM type after the name of the machine ?

There was no mockup so i put it there but i can move it. 
@ekidneyrh any preference where it should be located?

File: extensions/podman/src/podman-install.ts (hunk 3 above)
Comment (lstocchi) [reply extensions/podman/src/podman-install.ts]: Thanks, missed it. Done 👍

**Comment by jeffmaury** [comment]
> @ekidneyrh any preference where it should be located?

After the name looks good to me as it is also related to the custom image path so they will be close one to the other

**Comment by benoitf** [comment]
I'm wondering if we should not add a markdownDescription to the field of selection for vmType

Also I'm not sure if for an end user, appleHV or libkrun is meaningful. I know it's the value of the provider of podman, but do we need to expose that first or something like `default (Apple HyperVisor)` and `GPU passthrough (LibKrun)` or something else

**Comment by benoitf** [comment]
also, should it be the first option if it's not clear for users ? or it's a more advanced parameter (bottom option)

**Comment by lstocchi** [comment]
@benoitf @jeffmaury updated by following Florent suggestion. You can see the new labels in the video in the first post.

@jeffmaury i kept the name in the same position as you said, let me know what you think.

**Review by benoitf — APPROVED** [review]

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**